### PR TITLE
vrc-get: 1.5.2 -> 1.8.0

### PR DESCRIPTION
**Pull request by anatawa12**
## Description of changes

This PR upgrades vrc-get to 1.8.0
Obsolates #285844

CHANGELOG: https://github.com/vrc-get/vrc-get/blob/master/CHANGELOG.md#180---2024-03-31

I had never used the nix / nixos.
One user of vrc-get ask me about one bug that is fixed in the latest release.
I asked them to upgrade package, they told me that on their package manager it's latest and they are using nixpkgs.
So, I created this pull request to upgrade the vrc-get.

This PR switches download source from github to crates.io.
The vrc-get repository contains vrc-get-gui and vrc-get-litedb, which are not used by vrc-get so I switched to crates.io to reduce download size.

In addition, this PR removes openssl from dependencies since vrc-get uses rustls by default for TLS implementation since version 1.3.1.

I could not found documentation to get the suitable `hash` of `fetchCrate` and `cargoHash` so I ran `nix-build -A vrc-get` and get new hash from the error message.
Could you let me know what's the best way to get hashes?

If nixpkgs team allows, I want to update nixpkgs in my deployment CD (on github actions) and add to README.
Could you let me know if it is allowed to create pull requests automatically and preferred action / workflow for updating nixpkgs distribution?

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by phip1611 — APPROVED** [review]

File: pkgs/tools/misc/vrc-get/default.nix [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
+    hash = "sha256-+xbHw1DpFmapjsFoUvxUqTok8TKMebMw3gYjO/rx/iU=";
   };
 
   nativeBuildInputs = [ pkg-config ];
 
-  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ Security SystemConfiguration ];
+  buildInputs = lib.optionals stdenv.isDarwin [ Security SystemConfiguration ];
 
   # Make openssl-sys use pkg-config.
   OPENSSL_NO_VENDOR = 1;
```
Comment (bddvlpr) [inline pkgs/tools/misc/vrc-get/default.nix:17]:
```suggestion
```

**Review by bddvlpr — CHANGES_REQUESTED** [review]
Thank you for the PR, I was stuck in nixpkgs hell with the previous PR that I haven't gotten around to update it (yet again) while still in PR.

LGTM, only an unused input arg.

**Comment by bddvlpr** [comment]
> I could not found documentation to get the suitable hash of fetchCrate and cargoHash so I ran nix-build -A vrc-get and get new hash from the error message.
Could you let me know what's the best way to get hashes?

If using the new `nix-commands`, [nix hash](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-hash), else [nix-hash](https://nixos.org/manual/nix/stable/command-ref/nix-hash.html). 

> If nixpkgs team allows, I want to update nixpkgs in my deployment CD (on github actions) and add to README.
Could you let me know if it is allowed to create pull requests automatically and preferred action / workflow for updating nixpkgs distribution?

Nixpkgs already has a solution for that, [nixpkgs-update](https://ryantm.github.io/nixpkgs-update/).

File: pkgs/tools/misc/vrc-get/default.nix [hunk 2]
```diff
@@ -1,28 +1,26 @@
-{ fetchFromGitHub, lib, rustPlatform, pkg-config, openssl, stdenv, Security, SystemConfiguration }:
+{ fetchCrate, lib, rustPlatform, pkg-config, openssl, stdenv, Security, SystemConfiguration }:
```
[the remark this answers is not in the document]
Comment (anatawa12) [reply pkgs/tools/misc/vrc-get/default.nix]: fixed in adae70b6

File: pkgs/tools/misc/vrc-get/default.nix (hunk 1 above)
Comment (anatawa12) [reply pkgs/tools/misc/vrc-get/default.nix]: fixed in adae70b6

**Review by bddvlpr — APPROVED** [review]

**Review by bddvlpr — APPROVED** [review]

**Comment by anatawa12** [comment]
> Nixpkgs already has a solution for that, [nixpkgs-update](https://ryantm.github.io/nixpkgs-update/).

Thank you. I'll try it

**Comment by willbush** [comment]
> > Nixpkgs already has a solution for that, [nixpkgs-update](https://ryantm.github.io/nixpkgs-update/).
> 
> Thank you. I'll try it

@anatawa12 I was just perusing nixpkgs-update issues (for some reason) saw your issue https://github.com/nix-community/nixpkgs-update/issues/423 and I think you might have misunderstood what @bddvlpr meant. nixpkgs-update is [run as a bot](https://nix-community.org/update-bot/) that automatically updates packages. You shouldn't have to use it directly via a Github action.

**Comment by anatawa12** [comment]
@willbush Thank you for correction. I was planned to execute nixpkgs-update in release workflow on GitHub actions.

If nixpkgs have bot to upgrade, What was happening for vrc-get 1.6.x and 1.7.x? nixpkgs was stacked with 1.5.2 for several months

**Comment by anatawa12** [comment]
Sorry I did not notice that comment was updated. I saw the log file and the old manifest file was building litedb module which requires dotnet but and not required for vrc-get with default features. In this pr, vrc-get is not downloaded from github so litedb module is not built. Thank you for providing information.

https://nixpkgs-update-logs.nix-community.org/vrc-get/2024-04-06.log

**Comment by willbush** [comment]
@anatawa12 here are the logs https://nixpkgs-update-logs.nix-community.org/vrc-get/ Looks like something to do with `failed to run custom build command for `vrc-get-litedb v0.1.4-beta.0 (/build/source/vrc-get-litedb)` not sure :thinking: Do you recognize it?

**Comment by willbush** [comment]
@anatawa12 Also, Thanks for contributing to nixpkgs! :heart: Please consider [becoming a maintainer](https://github.com/NixOS/nixpkgs/tree/master/maintainers#how-to-become-a-maintainer) of your package. :)

**Comment by anatawa12** [comment]
I'm a just a developer of vrc-get and not using nixpkgs. I'll add me to maintainer after the nixpkgs-update command become working in my environment. 😄

**Comment by willbush** [comment]
@anatawa12 replied on the issue. I posted a way for you to run it in the docker container.

**Pull request merged**